mailpoet/lib/Automation/Integrations/WordPress/Subjects/UserSubject.php

```diff
@@ -0,0 +1,97 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\WordPress\Subjects;
+
+use DateTimeImmutable;
+use MailPoet\Automation\Engine\Data\Field;
+use MailPoet\Automation\Engine\Data\Subject as SubjectData;
+use MailPoet\Automation\Engine\Integration\Payload;
+use MailPoet\Automation\Engine\Integration\Subject;
+use MailPoet\Automation\Engine\WordPress;
+use MailPoet\Automation\Integrations\WordPress\Payloads\UserPayload;
+use MailPoet\Validator\Builder;
+use MailPoet\Validator\Schema\ObjectSchema;
+use WP_User;
+
+/**
+ * @implements Subject<UserPayload>
+ */
+class UserSubject implements Subject {
+  const KEY = 'wordpress:user';
+
+  /** @var WordPress */
+  private $wordPress;
+
+  public function __construct(
+    WordPress $wordPress
+  ) {
+    $this->wordPress = $wordPress;
+  }
+
+  public function getName(): string {
+    return __('WordPress user', 'mailpoet');
+  }
+
+  public function getKey(): string {
+    return self::KEY;
+  }
+
+  public function getArgsSchema(): ObjectSchema {
+    return Builder::object([
+      'user_id' => Builder::integer()->required(),
+    ]);
+  }
+
+  public function getPayload(SubjectData $subjectData): Payload {
+    $id = $subjectData->getArgs()['user_id'];
+    $user = new WP_User($id);
+    return new UserPayload($user);
+  }
+
+  public function getFields(): array {
+    global $wp_roles;
+    $roles = [];
+    foreach ($wp_roles->role_names as $id => $name) { // phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+      $roles[] = ['id' => $id, 'name' => $name];
+    }
+
+    return [
+      new Field(
+        'wordpress:user:email',
+        Field::TYPE_STRING,
+        __('Email', 'mailpoet'),
+        function (UserPayload $payload) {
+          return $payload->getEmail();
+        }
+      ),
+      new Field(
+        'wordpress:user:is-guest',
```

Comment:
What is a WordPress User Guest? I understand the `exist()` boolean, but there is no such concept in WordPress at the end, right? Not sure if `wordpress:user:exists` makes sense for a field. But `guest` makes to me only sense in WooCommerce atm.

Reply:
@websupporter Isn't it more the other way round — WP has guest users, but Woo doesn't? What I mean is that in WP you have a user with ID `0` and the `exists()` method, but in Woo doesn't seem to have such a concept for customers.

I guess you're asking about the naming of this field? In any way, this is a question for @NeosinneR, as this is how it was specified.

Reply:
The idea behind is `Is guest` field is that you can have special automation for WooCommerce guest customers. E.g., if a guest finishes order, you can start an automation to let them know they can register and automatically get 5% discount or similar. Or share what benefits registering offers. 

So yeah, this is targeted toward WC rather than WP, but I'll let you two decide on what makes sense in the code.